src/main/kotlin/no/digdir/informasjonsforvaltning/fdk_dataset_harvester/fuseki/HarvestFuseki.kt

```diff
@@ -36,21 +44,86 @@ class HarvestFuseki(private val fusekiProperties: FusekiProperties) {
             }
         }
 
+    fun queryAsk(query: String): Boolean =
+        datasetConnection().use {
+            it.begin(ReadWrite.READ)
+            try {
+                it.queryAsk(query)
+            } catch (ex: Exception) {
+                LOGGER.error("sparql-ask exception: $ex")
+                false
+            }
+        }
+
+    fun queryConstruct(query: String): Model? =
+        datasetConnection().use {
+            it.begin(ReadWrite.READ)
+            try {
+                it.queryConstruct(query)
+            } catch (ex: Exception) {
+                LOGGER.error("sparql-construct exception: $ex")
+                null
+            }
+        }
+
     fun queryDescribe(query: String): Model? =
         datasetConnection().use {
             it.begin(ReadWrite.READ)
             return try {
                 it.queryDescribe(query)
             } catch (ex: Exception) {
-                LOGGER.error("sparql exception: $ex")
+                LOGGER.error("sparql-describe exception: $ex")
                 null
             }
         }
 
+    fun querySelect(query: String, format: QueryResponseFormat): String? =
+        datasetConnection().use {
+            it.begin(ReadWrite.READ)
+            var result: String? = null
+            try {
+                it.queryResultSet(query) {
+                    qs ->
+                        result = when (format) {
+                            QueryResponseFormat.TEXT -> ResultSetFormatter.asText(qs).valuesOrNull(format)
+                            QueryResponseFormat.JSON -> querySelectWithJsonResponse(qs)?.valuesOrNull(format)
+                            QueryResponseFormat.XML -> ResultSetFormatter.asXMLString(qs).valuesOrNull(format)
+                    }
+                }
+            } catch (ex : Exception) {
+                LOGGER.error("sparql-select exception: $ex")
+                throw ex
```

Comment:
Kunne potensielt hatt en custom-exception her, og på de andre sparql-metodene, fange de i kontrolleren og svare med 4** i stedet for 500 eller no-content

Reply:
400 sendes tilbake for feil i sparql syntaks og og rosøk på å bruke sparql update :) Er det noen andre tilfeller som burde ha 4xx response?

Reply:
Når det feiler her feks. Dette ender vel i en 500, men er vel strengt tatt en brukerfeil om det er feil i sparql-query

Reply:
Eller er det her QueryExceptionHTTP har jena kastet QueryExceptionHTTP her?